waiter/src/waiter/scheduler.clj

```diff
@@ -406,7 +406,12 @@
            service->service-instances' {}]
       (if-not service
         service->service-instances'
-        (let [active-instances (doall (map async/<!! active-instances))]
+        (let [start-time-ms (-> (t/now) .getMillis)
```

Comment:
We have a timer called `app->available-tasks`.  It's not the greatest name, but why don't we just consult the timer data?

Reply:
The main reason is the logging per service lets us get incremental logging updates for slow services vs the `app->available-tasks` timer which is for all services at once. I will switch to introducing a new metric per service and a new timer function like `start-stop-time!` to log when it is too slow.

Reply:
I removed the code and added a timeout to the do-health-checks call and add a log if it takes too long.